Makefile

```
@@ -214,37 +214,8 @@ test: generate
 ## Lint
 ##
 
-# These are escape newlines, looks super weird. Allows these to run in
-# parallel with `make -j`
-lint: \
-	lint-go-deadcode \
-	lint-misspell \
-	lint-go-vet \
-	lint-go-nakedret \
-	lint-go-fmt
-
-lint-go-deadcode: deps-go
-	deadcode cmd/ pkg/
-
-lint-misspell: deps-go
-	git ls-files \
-	  | grep -v pkg/simulator/testdata/bad_symlink \
-	  | grep -v assets.go$ \
-	  | xargs misspell -error -f 'misspell: {{ .Filename }}:{{ .Line }}:{{ .Column }}:corrected {{ printf "%q" .Original }} to {{ printf "%q" .Corrected }}'
-
-lint-go-vet:
-	go vet ./cmd/... ./pkg/...
-
-lint-go-nakedret: deps-go
-	nakedret ./pkg/... ./cmd/...
-
-# This is ugly. since go-fmt doesn't have a simple exit code, we use
-# some make trickery to handle failing if there;s output.
-lint-go-fmt: $(foreach c,$(shell gofmt -l ./pkg/ ./cmd/ | grep -vE 'assets.go|bindata.go'),fmt-fail/$(c))
-lint-go-fmt: deps-go
-fmt-fail/%:
-	@echo fmt failure in: $*
-	@false
+lint:
+	golangci-lint -j3 run
```

`make lint` no longer works (unless I install golangci-lint). This isn't _necessarily_ terrible, but it'd be nice to address it.

Installing is easy for macOS (could just add `brew install golangci-lint` to the `deps` make target), but the recommended way for windows/linux is to curl a binary.

I tried adding `_ "github.com/golangci/golangci-lint/"` to https://github.com/kolide/launcher/blob/seph/modern-lint/pkg/tools/tools.go#L13, but get an error that golangci-lint is a program, not an importable package. 

what _did_ work is: 
```suggestion
	golangci-lint -j3 run
	go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.43.0
```

I'm not sure how you feel about that pattern to the makefile though.

---

Somewhat intentionally, I left it out of things like `tools` and the makefile. It feels like the install should be _outside_ this.  In some ways, I'm stepping back from whatever I advocate from when I wrote this Makefile. But it feels like the world mostly moved to using this tool, and I don't want to fight it.

Might be reasonable to throw an error from `make` though. (Though make's normal command not found is understandable enough. Ish)

---

If it's an intentional choice that's fine, but IMO it's a step backward in the dev experience for the project, I always loved how seamlessly `make deps ; make launcher`  worked. 

On the other hand, I don't see a clearly idiomatic way to install this without `brew` or `curl .. | sh`, so.. ¯\\_(ツ)_/¯
(the `go install` above is not recommended by the golangci-lint project)

---

`make deps launcher` should still work, `lint` has always been an optional step.

But, it's also never worked without installing `go` out of band. Feels kinda the same.

---

good enough for me